cpp/oneapi/dal/backend/transfer.hpp

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/array.hpp"
+#include "oneapi/dal/backend/memory.hpp"
+
+namespace oneapi::dal::backend {
+
+#ifdef ONEDAL_DATA_PARALLEL
+template <typename T>
+inline std::tuple<array<T>, sycl::event> to_device(sycl::queue& q, const array<T>& ary) {
+    if (!ary.get_count()) {
```

Comment:
Not sure it is not a mistake if we transfer empty array

Reply:
Is there use case when it leads to the problem?

Reply:
I think no - but I'd rather put an assert instead of this branch